src/marqo/tensor_search/tensor_search.py

```diff
@@ -239,6 +239,10 @@ def add_documents_orchestrator(
         config: Config, add_docs_params: AddDocsParams,
         batch_size: int = 0, processes: int = 1,
     ):
+    
+    # Default device calculated here and not in add_documents call
+    if add_docs_params.device is None:
+        add_docs_params.device = utils.read_env_vars_and_defaults("_MARQO_BEST_AVAILABLE_DEVICE")
 
```

Comment:
As discussed in slack, this would force a breaking change to py-marqo and a larger systematic change if we did it at the higher, request level. For now this is an improvement on the current situation.

Reply:
Also yes, I don't believe we want to underscore our marqo variables

Reply:
I just put in the underscore to indicate it was an env var for internal use only, but I can remove it